### DISCUSSION TITLE
0.2.1

### DISCUSSION THREAD

**Discussion by kubilus1** [kubilus1/autoortho#20] [Announcements]
This release brings a few significant changes.  In particular, ortho packs are now provided via the setup and configuration screen.

If you do have custom or other existing scenery in a `Custom Scenery/z_autoortho` directory, it's important that you clean this up before attempting to use this update.  

This release also switches the texture file processing tooling to ISPC which is around twice as fast for most DDS files.

## What's Changed
* Restructure code for better dependency separation by @kubilus1 in https://github.com/kubilus1/autoortho/pull/14
* Orthodownloader by @kubilus1 in https://github.com/kubilus1/autoortho/pull/15
* Fix how mipmap spanning reads work by @kubilus1 in https://github.com/kubilus1/autoortho/pull/16
* Add ispc as option for DXT compression by @kubilus1 in https://github.com/kubilus1/autoortho/pull/17
* Add scenery setup to GUI by @kubilus1 in https://github.com/kubilus1/autoortho/pull/18
* Tweaks and cleanup by @kubilus1 in https://github.com/kubilus1/autoortho/pull/19

**Full Changelog**: https://github.com/kubilus1/autoortho/compare/0.1.2...0.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/kubilus1/autoortho/releases/tag/0.2.0'>0.2.0</a>.</em>

**Comment by kubilus1** [comment]
Over time I'll be adding more scenery packs for Asia, Africa, and South America.

If and when Ortho4XP works correctly with XP12 water and other features, I'll look into creating separately scenery for XP12 specifically, but ATM these should work with either.  Granted XP12 still has some inefficiencies with how it handles custom scenery, so you may want to stick with XP11 for now.

**Comment by ivotaHR** [comment]
Thank you. I am going to try.

**Comment by ivotaHR** [comment]
I tested Euro pack and  XP12. Worked great.

**Comment by kubilus1** [comment]
Excellent!  Thanks for the feedback!

**Comment by betampona** [comment]
Hello, trying with version 0.2.0
For the download on LFPO it takes 8 minutes 15, with the cache loaded on LFPO 3 minutes 24.
With previous versions it was 5 minutes and with the cache loaded less than a minute, see my reports in previous threads.
I made a flight from LFMN to LFPO FL360 speed 445KTS every 9 minutes I have the appearance of blue tiles on the side which then disappears.
Arrived on Paris LFPO to see the photo in descent to 12000 feet, it seems to me that the following tile is not of the orthophoto to see the repetitions of reasons between the 2 red lines.
Greetings.

Translated with www.DeepL.com/Translator (free version)

![b738 - 2022-12-09 14 33 39](https://user-images.githubusercontent.com/97102929/206735288-25c605e7-219a-4874-88a8-03b5bae63c97.png)

**Comment by kubilus1** [comment]
Blue lines are simply a DSF file that is not fully loaded.  I had to change
the minimum zoom level from 11 to 12 which probably accounts for the extra
time, this is mostly due to using bing maps, so maybe this can be tweaked
in the future.

Red lines do indicate a problem though, do you have any screenshots of that?

-Matt

**Comment by betampona** [comment]
Hello, I have just made a new flight LFPO EDDB at FL 380 and 445 Kts.
Before starting I reset the cache, I confirm that your tile +48+002 is not good in orthophoto (see previous message) for the rest of the flight until Berlin no orthophoto problem.
A lot of blue tiles on a larger surface because in the west - east direction the tiles are less wide than high so they are renewed more often.

A question that may seem stupid to you, to increase the download speed and thus decrease the time, would it not be necessary to ask 2 requests to the Bing server using for example the even and odd tiles.
Greetings

Translated with www.DeepL.com/Translator (free version)

**Reply by kubilus1** [reply]
It is possible that the source maps themselves are not good.  If it is Bing maps as the source (BI) go to the website and see if it looks okay there.  You may need to zoom in and out a bit to get the same photos.   I've definitely seen some patches of Bing maps that are not so nice.

You could try using a different source "Map override type" to maybe 'Firefly' and see if that looks better.

**Comment by ivotaHR** [comment]
![Untitled](https://user-images.githubusercontent.com/96768739/206850992-ad5cfbf9-9654-4d48-97cd-b41f4b4c14c2.png)

Excellent.

**Reply by betampona** [reply]
Hello ivotaHR, to switch from version 0.2.0 to version 0.2.1 do we have to reset everything or do we keep z_autoortho and the scenery in custom scenery.
Hi there.

**Reply by ivotaHR** [reply]
I just overwrite all files in my first created folder.

**Reply by betampona** [reply]
Alright I'll erase everything and rebuild.

**Reply by kubilus1** [reply]
You can keep all your existing ortho downloads between versions.  I save most of the data outside of the program directory so that it can be preserved between versions.  New versions will detect this.

It's possible in the future there will be a major structural change, but I'll try to avoid those as much as I can.

**Comment by betampona** [comment]
Version 0.2.1, download test on LFST 12 minutes 20 for a first download and 4 minutes 25 if the cache is full.
Try it yourself on LFST and tell me how long it takes you?
With version 0.1.0 it took me 5 minutes with the cache less than a minute

**Comment by kubilus1** [comment]
So possibly the longer time may be due to the much larger size of the
scenery directories.  When it was with the earlier version, was this a
smaller set of tiles?

There are some inefficiencies with how XPlane parsers terrain files that is
worse with the larger the directories.  This is part of the reason I split
these into multiple directories like z_eur_00, z_eur_01 etc.

Perhaps we'll come up with some ways to make this more efficient in the
future, but not sure right now.

**Comment by betampona** [comment]
Hello, I just made a download comparison between version 0.2.1 and version 0.1.0
For version 0.2.1 on YSCB in Australia
initial download 5 minutes 35
with the cache filled 2 minutes 23
For version 0.1.0 on YSCB for ortho I took your z_ortho_aus_pac_BI16 folder it is compiled and its value is 6.65 GB
initial download 6 minutes 17
with cache filled 48 seconds
Conclusion: for the initial download version 0.2.1 is 42 seconds faster
for the download with filled cache version 0.1.0 is faster by 1 minute 35, the jpeg image converter is 3 times faster than in version 0.2.1

**Reply by kubilus1** [reply]
No the jpeg convert is about twice as fast in 0.2.1 than 0.1.0.  The deal is in 0.1.0 it it allowing a smaller minimum zoom level, so it's converting less overall (4 times less).  There are some other shortcuts in 0.1.0 that while slightly faster cause various problems.

Here's the data when you compare compression in isolation: 

```
Linux detected
PYDDS MM0 STB: total 73.74726496898802  per 2.458242165632934
PYDDS MM1 STB: total 42.125524493996636  per 1.404184149799888
PYDDS MM3 STB: total 22.776042024997878  per 0.7592014008332626
PYDDS NOMM STB: total 31.59957800999109  per 1.0533192669997031
PYDDS MM0 ISPC: total 36.500960870995186  per 1.2166986956998396
PYDDS MM1 ISPC: total 29.631223737000255  per 0.9877074579000085
PYDDS MM3 ISPC: total 20.83591944500222  per 0.6945306481667406
PYDDS NOMM ISPC: total 6.1135630659991875  per 0.20378543553330625
NVCOMPRESS: total 89.62328645700472  per 2.9874428819001575
NVCOMPRESS NOMM: total 61.63088581900229  per 2.0543628606334097
```

This is on my personal laptop and the exact numbers will vary, but have the same basic results.  STB is the old compressor, and ISPC is the new one.  NVCOMPRESS is an external tool.   

Of particular note is the speed difference between the full compression of all mipmaps (MM0 tests) which is twice as fast for STB.  But when we show a test that only compresses the top image without mipmaps, ISPC is nearly 5 times faster!

**Comment by betampona** [comment]
Hello, for the Run.dat can't we have a custom icon with your colours to put on the desktop shortcut.

**Comment by betampona** [comment]
I have just passed from 80° which was my initial setting to 60° in the graphic settings of X-plane11, I find that it greatly improves the visual generated by autoortho, I made a test with the empty cache on Camberra at FL300 with the zibo, no problem of blue tile for a speed lower than 360 Kts above appearances of blue tiles by intermittent. With the zibo in cruise the GS speed depending on the wind can go up to 520 Kts

Translated with www.DeepL.com/Translator (free version)
